firebase-perf/src/test/java/com/google/firebase/perf/application/FragmentStateMonitorTest.java

```diff
@@ -0,0 +1,173 @@
+
+  @Test
+  public void lifecycleCallbacks_logFragmentScreenTrace() {
+    FragmentStateMonitor monitor =
+        new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, fma);
+    monitor.onFragmentResumed(mockFragmentManager, mockFragment);
+    verify(mockTransportManager, times(0)).log(any(TraceMetric.class), any());
+
+    monitor.onFragmentPaused(mockFragmentManager, mockFragment);
+    verify(mockTransportManager, times(1)).log(any(TraceMetric.class), any());
+
+    monitor.onFragmentResumed(mockFragmentManager, mockFragment);
+    verify(mockTransportManager, times(1)).log(any(TraceMetric.class), any());
+
+    monitor.onFragmentResumed(mockFragmentManager, mockFragment);
+    verify(mockTransportManager, times(2)).log(any(TraceMetric.class), any());
+  }
+
+  @Test
+  public void lifecycleCallbacks_cleansUpMap_duringActivityTransitions() {
+    // Simulate call order of activity + fragment lifecycle events
+    AppStateMonitor appStateMonitor =
+        spy(new AppStateMonitor(mockTransportManager, clock, configResolver, fma));
+    FragmentStateMonitor fragmentMonitor =
+        new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, fma);
+    doReturn(true).when(appStateMonitor).isScreenTraceSupported();
+    WeakHashMap<Fragment, Trace> map = fragmentMonitor.getFragmentToTraceMap();
+    // Activity_A onCreate registers FragmentStateMonitor, then:
+    appStateMonitor.onActivityStarted(mockActivity);
+    Assert.assertEquals(0, map.size());
+    appStateMonitor.onActivityResumed(mockActivity);
+    fragmentMonitor.onFragmentResumed(mockFragmentManager, mockFragment);
+    Assert.assertEquals(1, map.size());
+    appStateMonitor.onActivityPaused(mockActivity);
+    fragmentMonitor.onFragmentPaused(mockFragmentManager, mockFragment);
+    Assert.assertEquals(0, map.size());
+    appStateMonitor.onActivityPostPaused(mockActivity);
+    // Activity_B onCreate registers FragmentStateMonitor, then:
+    appStateMonitor.onActivityStarted(mockActivityB);
+    appStateMonitor.onActivityResumed(mockActivityB);
+    fragmentMonitor.onFragmentResumed(mockFragmentManager, mockFragment);
+    appStateMonitor.onActivityStopped(mockActivity);
+    Assert.assertEquals(1, map.size());
+  }
+
+  @Test
+  public void fragmentTraceCreation_truncatesName_whenFragmentNameTooLong() {
+    AppStateMonitor appStateMonitor =
+        spy(new AppStateMonitor(mockTransportManager, clock, configResolver, fma));
+    FragmentStateMonitor fragmentMonitor =
+        spy(new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, fma));
+    doReturn(true).when(appStateMonitor).isScreenTraceSupported();
+    doReturn(longFragmentName)
+        .when(fragmentMonitor)
+        .getFragmentScreenTraceName(nullable(Fragment.class));
+
+    fragmentMonitor.onFragmentResumed(mockFragmentManager, mockFragment);
+    fragmentMonitor.onFragmentPaused(mockFragmentManager, mockFragment);
```

Comment:
What are you checking against? Did I miss anything?

Reply:
No you didn't miss anything, it's still WIP

Reply:
Finished adding checks now.